### ISSUE TITLE
features not working?

### ISSUE THREAD

**Issue by HDVinnie** [Swizards/QuickBox#58]
Im not sure if I did something wrong or not but install went fine and setdisk worked. Heres my issue. On panel you can see its says Rtorrent disabled and iRSSi-Autodl yet under service controller they are set to on.

Next issue is that the bandwidth graph is not working....just blank. 

Also quote shows size i set it to but is not showing what is used. Same when in rutorrent. shows 0% and when hovered over says 7.9TB/7.9TB like its full when its not?

http://ultraimg.com/images/wXc6nAk.png
http://ultraimg.com/images/V5SOJgX.png
http://i.imgur.com/re7Gn3e.png?3

**Comment by HDVinnie** [comment]
PS this script is awesome and its really awesome you guys made it open source!

**Comment by JMSDOnline** [comment]
Hello @HDVinnie, 

Yes, this is a known issue with any server using **/home** as the primary mount partition.

The solution would be to reset quotas and assign them to that particular mount as quotas within the script are mounted to **/**

This fix isn't too terribly complicated and will be built into a future version.

In the meantime, try this out:

```
quotaoff -a
service quota stop
mount -o remount /home
quotacheck -auMF vfsv1
quotaon -uv /home
service quota start
```

You will additionally need to make a quick adjustment to the diskspace plugin within rutorrent as follows:

Navigate to the file **action.php** at ``/srv/rutorrent/plugins/diskspace/action.php``

Change the lines:
```
$total = shell_exec("/usr/bin/sudo /usr/sbin/repquota -u / | /bin/grep ^".$quotaUser." | /usr/bin/awk '{printf $4*1024}'");
$free = shell_exec("/usr/bin/sudo /usr/sbin/repquota -u / | /bin/grep ^".$quotaUser." | /usr/bin/awk '{printf ($4-$3)*1024}'");
```

-to-

```
$total = shell_exec("/usr/bin/sudo /usr/sbin/repquota -u /home | /bin/grep ^".$quotaUser." | /usr/bin/awk '{printf $4*1024}'");
$free = shell_exec("/usr/bin/sudo /usr/sbin/repquota -u /home | /bin/grep ^".$quotaUser." | /usr/bin/awk '{printf ($4-$3)*1024}'");
```

For good measure, let's restart quotas with ``service quota restart`` followed by ``service apache2 restart``

Now your quotas should be able to get the proper assignments on the proper mounts.

As per the meter showing incorrectly for the bandwidth... this is completely my mistake and I am going over this in the script currently. Ensure that the proper network port your machine utilizes it's connection on is listed in these locations

1. /srv/rutorrent/home/widgets/stat.php -- $interface
2. /srv/rutorrent/home/widgets/data.php -- $int <-- at the top & echo '{"label":"  <-- at the bottom
3. /srv/rutorrent/home/widgets/config.php -- $iface_list & $iface_title
4. /srv/rutorrent/home/inc/config.php -- $interface

**Comment by HDVinnie** [comment]
Wow talk about great support for a open source project. Thank you and I will try this tonight and report back.

**Comment by HDVinnie** [comment]
[root@blah]:(829.6kb)~# quotaoff -a
[root@blah]:(829.6kb)~# quotaoff -a
[root@blah]:(829.6kb)~# service quota stop
[root@blah]:(829.6kb)~# mount -o remount /home
[root@blah]:(829.6kb)~# quotacheck -auMF vfsv1
[root@blah]:(829.6kb)~# quotaon -uv /home
quotaon: Mountpoint (or device) /home not found or has no quota enabled.
[root@blah]:(829.6kb)~# service quota start
[root@blah]:(829.6kb)~# 

I then ran 

nano /srv/rutorrent/plugins/diskspace/action.php and made the changes above and saved. That being said I think quotaon -uv /home failed?

**Comment by HDVinnie** [comment]
do i need to delete all the user accounts first? remake them all with new setdisk quotas ?

**Comment by JMSDOnline** [comment]
Try to run it again, I think I left out an important step in my Saturday booze haze. :stuck_out_tongue_winking_eye: 

```
quotaoff -a
service quota stop
sed -i 's/errors=remount-ro/usrquota,errors=remount-ro/g' /etc/fstab
mount -o remount /home
quotacheck -auMF vfsv1
quotaon -uv /home
service quota start
service apache2 restart
```

You should not have to remove any accounts as it should pickup the aquota file that is adjusted per the command ``setdisk``

Let me know how that goes and we'll move forward from there. If it continues to act up and you feel comfortable with it, you can email me your server details and I can login and review the issues first hand... we'll use that a last resort tactic.

**Comment by JMSDOnline** [comment]
After you have done that, run ``repquota -a`` and let me know those results.

**Comment by HDVinnie** [comment]
[root@blah]:(829.6kb)~# quotaoff -a
[root@blah]:(829.6kb)~# service quota stop
emount-ro/g' /etc/fstabb)~# sed -i 's/errors=remount-ro/usrquota,errors=r 
[root@ blah]:(829.6kb)~# mount -o remount /home
[root@ blah]:(829.6kb)~# quotacheck -auMF vfsv1
[root@ blah]:(829.6kb)~# quotaon -uv /home
quotaon: Mountpoint (or device) /home not found or has no quota enabled.
[root@ blah]:(829.6kb)~# service quota start
[root@ blah]:(829.6kb)~# service apache2 restart
[root@ blah]:(829.6kb)~# repquota -a
*** Report for user quotas on device /dev/root
Block grace time: 7days; Inode grace time: 7days
                        Block limits                File limits
User            used    soft    hard  grace    used  soft  hard  grace
----------------------------------------------------------------------
root      -- 2926364       0       0          58087     0     0       
man       --    2204       0       0             71     0     0       
www-data  --   58448       0       0           3031     0     0       
systemd-timesync --       0       0       0              1     0     0       
syslog    --  228876       0       0             17     0     0       
bind      --       8       0       0              2     0     0       
vnstat    --      76       0       0             19     0     0       
main   --     424 8283750400 8283750400             81     0     0       
box1      --     100 2097152000 2097152000             23     0     0       
box2      --     228 2097152000 2097152000             46     0     0       
#501      --     144       0       0             16     0     0

**Comment by HDVinnie** [comment]
I don't want to be a pain so I can just wait for stable release that works with /home if need be

**Comment by JMSDOnline** [comment]
Will you do ``cat /etc/fstab`` for me real quick. Also, this is not a problem... it will help me further along in my conclusion for **/home** mounts

**Comment by HDVinnie** [comment]
[root@blah]:(829.6kb)~# cat /etc/fstab
# <file system>	<mount point>	<type>	<options>	<dump>	<pass>
/dev/md2	/	ext4	usrquota,usrquota,errors=remount-ro,relatime	0	1
/dev/md3	/home	ext4	defaults,relatime	1	2
/dev/sda4	swap	swap	defaults	0	0
/dev/sdb4	swap	swap	defaults	0	0
/dev/sdc4	swap	swap	defaults	0	0
proc		/proc	proc	defaults		0	0
sysfs		/sys	sysfs	defaults		0	0
devtmpfs	/dev	devtmpfs	rw	0	0

**Comment by JMSDOnline** [comment]
Ah, ok.. I see. 

Can you adjust the /etc/fstab to look as follows:

```
/dev/md2 / ext4 defaults,relatime 1 2
/dev/md3 /home ext4 usrquota,usrquota,errors=remount-ro,relatime 0 1
/dev/sda4 swap swap defaults 0 0
/dev/sdb4 swap swap defaults 0 0
/dev/sdc4 swap swap defaults 0 0
proc /proc proc defaults 0 0
sysfs /sys sysfs defaults 0 0
devtmpfs /dev devtmpfs rw 0 0
```

**Comment by HDVinnie** [comment]
done

now rerun-
quotaoff -a
service quota stop
sed -i 's/errors=remount-ro/usrquota,errors=remount-ro/g' /etc/fstab
mount -o remount /home
quotacheck -auMF vfsv1
quotaon -uv /home
service quota start
service apache2 restart

?

**Comment by JMSDOnline** [comment]
Just rerun this bit..

```
quotaoff -a
service quota stop
mount -o remount /home
quotacheck -auMF vfsv1
quotaon -uv /home
service quota start
```

**Comment by HDVinnie** [comment]
[root@blah]:(829.6kb)~# quotaoff -a
[root@ blah]:(829.6kb)~# service quota stop
[root@ blah]:(829.6kb)~# mount -o remount /home
[root@ blah]:(829.6kb)~# quotacheck -auMF vfsv1
[root@ blah]:(829.6kb)~# quotaon -uv /home
/dev/md3 [/home]: user quotas turned on
[root@ blah]:(829.6kb)~# service quota start
Job for quota.service failed because the control process exited with error code. See "systemctl status quota.service" and "journalctl -xe" for details.
[root@ blah]:(829.6kb)~#

**Comment by JMSDOnline** [comment]
ok, what does ``systemctl status quota.service`` show us - we're almost there. :smile:

**Comment by HDVinnie** [comment]
[root@blah]:(829.6kb)~# -l [root@ns527149]:(829.6kb)~# systemctl status quota.service
-bash: syntax error near unexpected token `('
[root@ blah]:(829.6kb)~# ● quota.service - Check And Enable File System Quotas
e: Main process exit...E
Apr 17 10:50:57 ns527149.ip-149-56-18.net systemd[1]: Failed to start Check And Enable....
Apr 17 10:50:57 ns527149.ip-149-56-18.net systemd[1]: quota.service: Unit entered fail....
Apr 17 10:50:57 ns527149.ip-149-56-18.net systemd[1]: quota.service: Failed with resul....
Hint: Some lines were ellipsized, use -l to show in full.●: command not found
[root@ blah]:(829.6kb)~#    Loaded: loaded (/lib/systemd/system/quota.service; enabled; vendor preset: enabled)
-bash: syntax error near unexpected token `('
[root@ blah]:(829.6kb)~#    Active: failed (Result: exit-code) since Sun 2016-04-17 10:50:57 EDT; 2min 20s ago
-bash: syntax error near unexpected token `('
[root@ blah]:(829.6kb)~#      Docs: man:quotaon(8)
-bash: syntax error near unexpected token `('
[root@ blah]:(829.6kb)~#   Process: 22435 ExecStop=/usr/share/quota/quotaoff.sh (code=exited, status=0/SUCCESS)
-bash: syntax error near unexpected token `('
[root@ blah]:(829.6kb)~#   Process: 16333 ExecStart=/usr/share/quota/quotaon.sh (code=exited, status=1/FAILURE)
-bash: syntax error near unexpected token `('
[root@ blah]:(829.6kb)~#  Main PID: 16333 (code=exited, status=1/FAILURE)
-bash: syntax error near unexpected token `('
[root@ blah]:(829.6kb)~#

**Comment by JMSDOnline** [comment]
Will you try ``/etc/init.d/quota start`` and see if this results in success?

**Comment by HDVinnie** [comment]
Starting quota (via systemctl): quota.serviceJob for quota.service failed because the control process exited with error code. See "systemctl status quota.service" and "journalctl -xe" for details.
 failed!

**Comment by JMSDOnline** [comment]
Yeah, it was a stretch. I am building a server now and I am going to mount /home. I will do some sampling and let you know what I find.

Although, I am curious... what if you tried to do ``setdisk`` on a user you already ran the command? Do not delete them, just run the command over them once more.

**Comment by HDVinnie** [comment]
[root@blah]:(829.6kb)~# setdisk
Username: Main
Quota size for user: (EX: 500GB): 
8000GB
[root@blah]:(829.6kb)~# 

Done

**Comment by JMSDOnline** [comment]
Wow, did it actually reflect as such on the users account?

**Comment by HDVinnie** [comment]
http://i.imgur.com/zsmiGwJ.png?1

**Comment by JMSDOnline** [comment]
Perfect! Your disk space is reflected properly. Now, as per the BW graph.

The following files need to be adjusted ...

1. /srv/rutorrent/home/widgets/stat.php -- $interface
2. /srv/rutorrent/home/widgets/data.php -- $int <-- at the top & echo '{"label":" <-- at the bottom
3. /srv/rutorrent/home/widgets/config.php -- $iface_list & $iface_title
4. /srv/rutorrent/home/inc/config.php -- $interface

looks like **teql0** is your primary network adapter. So you will need to replace **eth0** is the files I listed above.Not sure why these didn't update automagically, however, I am almost done with a new script install to test any of these inconsistencies.

**Comment by HDVinnie** [comment]
also in rutorrent

http://i.imgur.com/7r7wVp2.png?1

**Comment by HDVinnie** [comment]
but the widget for disk space still shows as 0% being used even though there is 25 torrents loaded? So it doesnt seem to be working

**Comment by JMSDOnline** [comment]
Yeah, I am looking closer at it. What is the provider of this server?

**Comment by HDVinnie** [comment]
OVH

**Comment by HDVinnie** [comment]
Storage FS-12T - 16G Atom C2750 SoftRaid 3x4 TB Server

**Comment by JMSDOnline** [comment]
Thank you, I am going to take a look at their setup.

**Comment by HDVinnie** [comment]
thanks for everything ill report back on bandwidth graph

**Comment by JMSDOnline** [comment]
I have created a patch for you... it's the disk_data plugin I created is repquoting the **/** mount. I have made the necessary adjustments.. this should now show as properly. 

Upload this file to ``/srv/rutorrent/home/widgets/`` - overwrite the disk_data.php currently there.

[disk_data.php patch file via Dropbox](https://www.dropbox.com/s/55glw8cwsjql1nl/disk_data.php?dl=0)

**Comment by HDVinnie** [comment]
Worked beautiful!

**Comment by JMSDOnline** [comment]
Glad to hear!

Looks like I have my work cut out for me then as to adding the **/home** option to the script. I really appreciate your patience and feedback through this process.

**Comment by HDVinnie** [comment]
no thank you! teql0 did not work but I will go through the list of them and try them all lol....unless you know a command or something i can run to find out

**Comment by JMSDOnline** [comment]
Try ``ifstat`` and see which network adapter shows up as busy when you add a torrent.

**Comment by HDVinnie** [comment]
eth0 LOL

going to redo it again with eth0 and report back

**Comment by JMSDOnline** [comment]
Gotta love OVH....

**Comment by HDVinnie** [comment]
lol well still a no go changing

/srv/rutorrent/home/widgets/stat.php -- $interface
/srv/rutorrent/home/widgets/data.php -- $int <-- at the top & echo '{"label":" <-- at the bottom
/srv/rutorrent/home/widgets/config.php -- $iface_list & $iface_title
/srv/rutorrent/home/inc/config.php -- $interface

but least of my worries....thats just a cosmetic perk. Thanks for everything.

**Comment by HDVinnie** [comment]
eth0       
 KB/s in  KB/s out
 6179.15    561.27
 5494.68    508.29
 3321.28    410.68
 3581.73    538.78
 3875.83    484.09
 3936.53    536.08
 3653.28    591.26
 1796.15    533.79
 2145.64    522.47
 2354.60    466.56
 2371.46    473.06
 2413.00    458.05
 2334.96    431.57
 2337.69    387.75
 2331.51    348.54
 2653.17    405.49
 3296.01    400.11
 4067.75    483.45

Just incase it can help you with future patch/builds

**Comment by JMSDOnline** [comment]
It certainly will help, again, I appreciate your reports. I am going to look into OVH and see what in their setups may be blocking a proper read on the charts.

I'll shoot some updates your way in the next couple of days.

**Comment by HDVinnie** [comment]
Thanks

**Comment by HDVinnie** [comment]
ok one more question on this topic.....would anything we did mess up the other two boxs I have on this server. As main is working fine. Box1 and Box2 are working but when launching rutorrent for Box1 and 2 i just get the 

[17.04.2016 14:21:35] WebUI started.
[17.04.2016 14:21:35] No connection to rTorrent. Check if it is really running. Check $scgi_port and $scgi_host settings in config.php and scgi_port in rTorrent configuration file.

So i restarted server (no change)
logged in SSH as root and ran reload (no change)
logged in via SSH with box1 and box2 deets and ran screen -fa -dmS rtorrent rtorrent and screen -fa -dmS irssi irssi (no change)

**Comment by JMSDOnline** [comment]
When the script was first committed I remember a few people had the issue and I have since resolved it, but for **/** mounts. This issue may be present in **/home** mounts still - as we have seen there are a few kinks to address.. am I right.. anyways.

Try as root to run it as this...

``sudo -u [username] screen -list``

Check the pid that is listed for both irssi and rtorrent. 

Then kill the process with ``kill -9 [process-id]``. 
Next, wipe the dead screen with ``sudo -u [username] screen -wipe``. 
Now wait about 30 seconds for the **./startup** to pickup the cron and reignite the process.

This should work without error, but again, I haven't done a lot of testing yet on **/home** mount installs. It's possible that since the quota is on the same mount as the data (etc etc) these processes may hiccup during the process... but to answer the question, what we did should not have any adverse effects... other than some crazy system burp.

**Comment by HDVinnie** [comment]
No go but no worries. Remade box1 and 2 since they were pretty much empty. Working now 

Thanks again

**Comment by JMSDOnline** [comment]
Well that's good. I'm glad they weren't too vested in data and all is well now. If it did cause issue, that is again something I would need to personally monitor to know for sure rather than guess work. All is well though as I am now negotiating an OVH test server (glad people just have these things lying around). I'll be able to run all sorts of tests and apply fixes.

Be on the lookout for many many patches in the coming days. :smile_cat:

**Comment by HDVinnie** [comment]
Cheers I got push notifications coming right to my iPhone : )
